src/main/java/uk/gov/hmcts/reform/coh/service/QuestionStateService.java

```diff
@@ -25,4 +26,9 @@ public QuestionState retrieveQuestionStateById(final Integer questionStateId){
     public Optional<QuestionState> retrieveQuestionStateByStateName(final String stateName){
         return questionStateRepository.findByState(stateName);
     }
+
+    public QuestionState fetchQuestionState(QuestionStates state) {
+        return retrieveQuestionStateByStateName(state.getStateName())
+            .orElseThrow(() -> new RuntimeException("Unknown question state: " + state.name()));
+    }
```

Comment:
A RuntimeException would kill the application?

Reply:
No, it wouldn't. I was choosing between `Optional.flatMap()` and this, and would be happy to switch over.

Reply:
... and fixed in da02c94. Thanks for spotting it!